templates/shared_services/admin-vm/terraform/admin-jumpbox.tf

```diff
@@ -51,7 +53,10 @@ resource "azurerm_windows_virtual_machine" "jumpbox" {
     disk_encryption_set_id = var.enable_cmk_encryption ? azurerm_disk_encryption_set.jumpbox_disk_encryption[0].id : null
   }
 
-  lifecycle { ignore_changes = [tags] }
+  # ignore changes to secure_boot_enabled and vtpm_enabled as these are destructive
+  # (may be allowed once https://github.com/hashicorp/terraform-provider-azurerm/issues/25808 is fixed)
+  #
+  lifecycle { ignore_changes = [tags, secure_boot_enabled, vtpm_enabled] }
```

Comment:
@jonnyry can you explain why we need all the ignore_changes as some of the templates don't allow upgrade of the images.
plus, if we really want to prevent this maybe we should version it with a "major" jump.
Just curious of the reasons.

Reply:
ok